src/voodoo/spellbook/spells/haste.ts

```diff
@@ -42,8 +42,9 @@ export const haste: SpellFunction = async (voodoo, accountId, upgradeConfigs) =>
 
   for (const playerId of playerIds) {
     const baseSpeed = await voodoo.getPlayerCheckStatBase({ accountId: playerId, stat: 'speed' });
+    const currentSpeed = await voodoo.getPlayerCheckStatCurrent({accountId: playerId, stat: 'speed' });
 
-    if (baseSpeed) {
+    if (baseSpeed == currentSpeed) {
```

Comment:
I'd say that if a user is already under the influence of a normal speed potion but has a Haste invocation that would make them go even faster, Voodoo would make up for the difference.

With this current proposed change, once under the influence of any other speed effect, Haste does not work _at all_.

I'd suggest this

```suggestion
    const buffedSpeed = baseSpeed * (1 + bonus);
    const speedDelta = buffedSpeed - currentSpeed;

    if (speedDelta > 0) {
```